src/network/networkd-dhcp6.c

```diff
@@ -835,7 +832,44 @@ static int dhcp6_set_unreachable_route(Link *link, const union in_addr_union *ad
 
         (void) set_remove(link->dhcp6_routes_old, ret);
 
-        return 1;
+        return 0;
+}
+
+static int log_dhcp6_pd_prefix(Link *link, const union in_addr_union *prefix, uint8_t prefixlen) {
+        _cleanup_free_ struct in_addr_prefix *p = NULL;
+        _cleanup_free_ char *buf = NULL;
+        int r;
+
+        assert(link);
+        assert(prefix);
+
+        p = new(struct in_addr_prefix, 1);
+        if (!p)
+                return log_oom();
+
+        *p = (struct in_addr_prefix) {
+                .family = AF_INET6,
+                .prefixlen = prefixlen,
+                .address = *prefix,
+        };
+
+        (void) in_addr_prefix_to_string(p->family, &p->address, p->prefixlen, &buf);
+
+        if (prefixlen > 64)
+                log_link_warning(link, "DHCP6: received PD Prefix %s with prefix length > 64, ignoring.",
```

Comment:
shouldn't it return here? or is it intended to add this to link->dhcp6_pd_prefixes regardless of the size?

Reply:
This is intended. The message should not be logged at warning level frequently.